differential/diffAtlas_DE_deseq.R

```diff
@@ -5,468 +5,120 @@
-
-	# Re-order the rows based on the assay names.
-	allBatches <- allBatches[ order( allBatches$AssayName ), ]
-
-	rownames( allBatches ) <- allBatches$AssayName
-	allBatches$AssayName <- NULL
-
-	return( allBatches )
-}
-
-
-# make_assays_to_bioreps_df
-# 	- Given a list of biological replicate objects create a data frame mapping
-# 	assay names to their biological replicate names. These will either be the
-# 	same as the assay name (no technical replicates), or the technical
-# 	replicate group ID (technical replicates).
-make_assays_to_bioreps_df <- function( bioReps ) {
-
-	assaysToBioRepsList <- lapply( bioReps, function( bioRep ) {
-
-		assayNames <- biorep_assay_names( bioRep )
-		techRepId <- technical_replicate_id( bioRep )
-
-		if( length( techRepId ) > 0 ) {
-			
-			data.frame( AssayName = assayNames, BioRepName = techRepId, stringsAsFactors = FALSE )
-
-		} else {
-			
-			data.frame( AssayName = assayNames, BioRepName = assayNames, stringsAsFactors = FALSE )
-		}
-	})
-
-	assaysToBioRepsDf <- do.call( "rbind", assaysToBioRepsList )
-
-	return( assaysToBioRepsDf )
-}
-
-
-# add_tech_rep_sums
-# 	- Given a data frame of expression values, the data frame of biological
-# 	replicate annotations, technical replicates group IDs, calculate the sum
-# 	for each set of technical replicate columns in the data frame, and replace
-# 	the original columns with the new column(s) containing the sums.
-add_tech_rep_sums <- function( expressionData, bioRepAnnotations, techRepGroupIDs ) {
-
-	# Create a list with one element per column of sums.
-	techRepSumsList <- lapply( techRepGroupIDs, function( techRepGroupID ) {
-
-		# Get the row indices for the technical replicates.
-		techRepRows <- which( bioRepAnnotations$BioRepName == techRepGroupID )
-
-		# Get the assay names on these rows.
-		techRepAssayNames <- bioRepAnnotations$AssayName[ techRepRows ]
-
-		# Get the columns of normalized data for these assays.
-		techRepColumns <- expressionData[ , techRepAssayNames ]
-
-		# Get the sum of each row.
-		techRepSums <- apply( techRepColumns, 1, function( x ) { sum( x ) } )
-
-		techRepSumsDF <- structure( data.frame( techRepSums ), names = techRepGroupID )
-	} )
-
-	# Turn the list into a data frame.
-	techRepSumsDF <- do.call( "cbind", techRepSumsList )
-
-	# Next need to remove the columns for the technical replicate assays from
-	# the normalized data frame.
-	# First get all the assay names we want to remove.
-	allTechRepRows <- which( bioRepAnnotations$BioRepName %in% techRepGroupIDs )
-
-	allTechRepAssays <- bioRepAnnotations$AssayName[ allTechRepRows ]
-
-	# Remove the columns for these assay names from the normalized data frame.
-	expressionData <- expressionData[ , !( colnames( expressionData ) %in% allTechRepAssays ), drop = FALSE ]
-
-	# Add the columns of sums instead.
-	expressionData <- cbind( expressionData, techRepSumsDF )
-
-	return( expressionData )
-}
 
+diffAtlas_DE_deseq2 <- function( expAcc, atlasProcessingDirectory ) {
```

Comment:
In the older strategy, this function had a 3rd parameter (countsMatrixFile). I would suggest to check that upstream code which provides 3 parameters is not breaking. I am thinking specifically about [this](https://github.com/ebi-gene-expression-group/atlas-analysis/blob/feature/updated_limma_strategy/differential/diffAtlas_DE.pl#L146).

Reply:
Good spot- it's fixed in the R script, but not the perl

Reply:
Actually no, correction, this is already addressed. The call at https://github.com/ebi-gene-expression-group/atlas-analysis/blob/0d49a372c97a2b71c0ba849b27ffffe75c445b96/differential/diffAtlas_DE.pl#L411 is corrected appropriately.